pkgs/applications/audio/musescore/default.nix

```diff
@@ -140,6 +140,26 @@ in stdenv'.mkDerivation (finalAttrs: {
     ln -s $out/Applications/mscore.app/Contents/MacOS/mscore $out/bin/mscore
   '';
 
+  # muse-sounds-manager installs Muse Sounds sampler libMuseSamplerCoreLib.so.
+  # It requires that argv0 of the calling process ends with "/mscore" or "/MuseScore-4".
+  # We need to ensure this in two cases:
+  #
+  # 1) when the user invokes MuseScore as "mscore" on the command line or from
+  #    the .desktop file, and the normal argv0 is "mscore" (no "/");
+  # 2) when MuseScore invokes itself via File -> New, and the normal argv0 is
+  #    the target of /self/proc/exe, which in Nixpkgs was "{...}/.mscore-wrapped"
+  #
+  # In order to achieve (2) we install the final binary as $out/mscore, and
+  # in order to achieve (1) we use makeWrapper without --inherit-argv0.
+  #
+  # wrapQtAppsHook uses wrapQtApp -> wrapProgram -> makeBinaryWrapper --inherit-argv0
+  # so we disable it and explicitly use makeQtWrapper.
+  dontWrapQtApps = stdenv.isLinux;
```

Comment:
Why is it an issue to wrapQtApps on non linux systems? Again please.

Reply:
After this PR:
1) On Linux, we disable the default `wrapQtAppsHook` and wrap the binary explicitly.
2) On macOS, nothing changes.

If I did this unconditionally:
```
  dontWrapQtApps = true;
  postFixup = ''
    mv $out/bin/mscore $out
    makeQtWrapper $out/mscore $out/bin/mscore
  '';
```
I might break MuseScore on macOS because there `wrapQtAppsHook` probably wraps not only `$out/bin/mscore` but also something in `$out/Applications/`, and users may need that.

This PR needs testing on macOS and possibly some changes to be enabled for macOS.

Reply:
macOS needs wrapQtAppsHook for the same reason Linux does: Qt has to find its plugins with the help of QT_PLUGIN_PATH and NIXPKGS_QT6_QML_IMPORT_PATH environment variables.

Reply:
> macOS needs wrapQtAppsHook for the same reason Linux does: Qt has to find its plugins with the help of QT_PLUGIN_PATH and NIXPKGS_QT6_QML_IMPORT_PATH environment variables.

That's what I thought, otherwise `wrapQtAppsHook` would have been set to `null` on Darwin platforms.